exonum/src/runtime/mod.rs

```diff
@@ -359,10 +373,34 @@ impl<'a> ExecutionContext<'a> {
     pub(crate) fn dispatch_action(&mut self, action: dispatcher::Action) {
         self.actions.push(action);
     }
+}
 
-    pub(crate) fn take_actions(&mut self) -> Vec<dispatcher::Action> {
-        let mut other = Vec::new();
-        std::mem::swap(&mut self.actions, &mut other);
-        other
+/// Instance descriptor contains information to access running service instance.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct InstanceDescriptor<'a> {
+    /// The current service instance identifier.
+    pub id: InstanceId,
+    /// The current service instance name.
+    pub name: &'a str,
+}
+
+impl std::fmt::Display for InstanceDescriptor<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}:{}", self.id, self.name)
     }
 }
+
+impl From<InstanceDescriptor<'_>> for (InstanceId, String) {
+    fn from(descriptor: InstanceDescriptor) -> Self {
+        (descriptor.id, descriptor.name.to_owned())
+    }
+}
+
+/// Change in list of service instances.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Ord, PartialOrd)]
```

Comment:
What is the purpose of deriving `Ord` and `PartialOrd`, as this struct can't be compared to others?